Src/DependencyCollector/Shared/HttpCoreDiagnosticSourceListener.cs

```diff
@@ -111,11 +117,20 @@ public void OnError(Exception error)
         /// <param name="evnt">The current notification information.</param>
         public void OnNext(KeyValuePair<string, object> evnt)
         {
-            const string ErrorTemplateTypeCast = "Event {0}: cannot cast {1} to expected type {2}";
-            const string ErrorTemplateValueParse = "Event {0}: cannot parse '{1}' as type {2}";
-
             try
             {
+                // It's possible to host multiple apps (ASP.NET Core or generic hosts) in the same process
+                // Each of this apps has it's own DependencyTrackingModule and corresponding Http listener.
+                // We should ignore events for all of them except one
+                if (!SubscriptionManager.IsActive(this))
+                {
+                    DependencyCollectorEventSource.Log.NotActiveListenerNoTracking(evnt.Key, Activity.Current?.Id);
```

Comment:
this is a part of a solution. It ensures subscription only happens once and picks one "victim" to send requests thru.

I think better solution will be to check application name in the listener and only process requests that belong to the application that created the subscriber

Reply:
All applications created subscribers. All of them are valid. Moreover, any listener could die at any moment and we still need to report telemetry through the next victim.

The limitation of this is that all applications have to have the same config/ikey. We discussed it briefly in the Microsoft/ApplicationInsights-aspnetcore#621 and everyone seems to be on board with it.

Reply:
Same configuration is a pretty heavy requirement if not for the nature of Service Fabric environment where you may want AI configuration out of the single host scope. This code, however, will run in non-service Fabric scenarios as well - do we have a scenario where a customer will encounter multiple listeners outside of SF? Will the requirement for same config still be fine in such a case?